src/ui/index.js

```diff
@@ -77,12 +87,74 @@ export default async (kbnServer, server, config) => {
       },
       vars: defaults(app.getInjectedVars() || {}, uiExports.defaultInjectedVars),
     };
-
+    if (kibanaTranslations.length <= 0) {
+      const locale = await getTranslationLocale(acceptLanguages);
+      kibanaTranslations = await i18nPlugin.getRegisteredLocaleTranslations(locale);
+    }
     return this.view(app.templateName, {
       app: app,
       loadingGif: loadingGif,
       kibanaPayload: payload,
       bundlePath: `${config.get('server.basePath')}/bundles`,
+      welcomeMessage: translate('CORE-WELCOME_MESSAGE'),
+      welcomeError: translate('CORE-WELCOME_ERROR'),
     });
   });
 };
+
+function translate(key) {
+  if (!(key in kibanaTranslations)) {
+    return null;
+  }
+  return kibanaTranslations[key];
+}
+
+async function getTranslationLocale(acceptLanguages) {
+  let localeStr = '';
+  let foundLang = false;
+
+  if (acceptLanguages === null || acceptLanguages.length <= 0) {
+    return DEFAULT_LOCALE;
+  }
+
+  const acceptLangsLen = acceptLanguages.length;
+  const registeredLocales = await i18nPlugin.getRegisteredTranslationLocales();
+
+  for (let indx = 0; indx < acceptLangsLen; indx++) {
+    const language = acceptLanguages[indx];
+    if (language.region) {
+      localeStr = language.code + '-' + language.region;
+    } else {
+      localeStr = language.code;
+    }
+    if (registeredLocales.indexOf(localeStr) > -1) {
+      foundLang = true;
+      break;
```

Comment:
Avoid using `break`, it makes the code more difficult to reason about. Also it would be preferable if these loops could be converted into a more functional style.